src/printer/smt2/smt2_printer.cpp

```diff
@@ -1907,15 +1919,74 @@ static void toStream(std::ostream& out,
          ++i)
     {
       const Datatype& d = i->getDatatype();
+      if (d.isParametric())
+      {
+        out << "(par (";
+        for (unsigned p = 0, nparam = d.getNumParameters(); p < nparam; p++)
+        {
+          out << (p > 0 ? " " : "") << d.getParameter(p);
+        }
+        out << ")";
+      }
       out << "(";
       toStream(out, d);
       out << ")";
+      if (d.isParametric())
+      {
+        out << ")";
+      }
     }
     out << ")";
   }
   else
   {
-    out << " () (";
+    out << " (";
+    // can only print if all datatypes in this block have the same parameters
+    // in theory, given input language 2.6 and output language 2.5, it could
+    // be impossible to print a datatype block where datatypes were given
+    // different parameter lists.
+    bool success = true;
+    const Datatype& d = datatypes[0].getDatatype();
+    unsigned nparam = d.getNumParameters();
+    for (unsigned j = 1, ndt = datatypes.size(); j < ndt; j++)
+    {
+      const Datatype& dj = datatypes[j].getDatatype();
+      if (dj.getNumParameters() != nparam)
+      {
+        success = false;
```

Comment:
Nit: Why not just break here and remove lines 1970-1973 (similar to the break statement on line 1966)?

Reply:
Because lines 1970-1973 also handle the case where the else branch set success to false.

Reply:
But that one `break`s anyway :)

Reply:
Right, so a break here is redundant :)

Reply:
I don't know if it was the lack of caffeine or something else but I misread the code and missed the fact that the other `break` was for the inner loop.